### PR TITLE
Fix: Mock Pub/Sub service in tests

### PR DESCRIPTION
**Pull request by cortadocodes**
### Fixes
- [x] Add `from_string` option to `Serialisable.deserialise`

### Testing
- [x] Mock Google Pub/Sub `Service`, `Topic`, `Subscription`, `Publisher` and `Subscriber` in tests
- [x] Remove unneeded cleanup code from `Service` tests

### CODE REVIEW

File: tests/cloud/pub_sub/mocks.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+MESSAGES = {}
+
+
+class MockTopic(Topic):
+    """A mock topic that registers in a global dictionary rather than Google Pub/Sub.
+
+    :param str name:
+    :param str namespace:
+    :param MockService service:
+    :return None:
```
Comment (thclark) [inline tests/cloud/pub_sub/mocks.py:14]: ??

File: tests/cloud/pub_sub/mocks.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+
+class MockSubscription(Subscription):
+    """A mock subscription that registers in a global dictionary rather than Google Pub/Sub.
+
+    :param str name:
+    :param MockTopic topic:
+    :param str namespace:
+    :param MockService service:
+    :return None:
```
Comment (thclark) [inline tests/cloud/pub_sub/mocks.py:39]: ??

File: tests/cloud/pub_sub/mocks.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        pass
+
+    def delete(self):
+        pass
+
+
+class MockFuture:
+    """A mock future that is returned when publishing or subscribing.
+
+    :return None:
```
Comment (thclark) [inline tests/cloud/pub_sub/mocks.py:52]: ??

File: tests/cloud/pub_sub/mocks.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        pass
+
+    def cancel(self):
+        pass
+
+
+class MockPublisher:
+    """A mock publisher that puts messages in a global dictionary instead of Google Pub/Sub.
+
+    :return None:
```
Comment (thclark) [inline tests/cloud/pub_sub/mocks.py:65]: ??

File: tests/cloud/pub_sub/mocks.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+    @staticmethod
+    def topic_path(project_name, topic_name):
+        return f"projects/{project_name}/topics/{topic_name}"
+
+
+class MockSubscriber:
+    """A mock subscriber that gets messages from a global dictionary instead of Google Pub/Sub.
+
+    :return None:
```
Comment (thclark) [inline tests/cloud/pub_sub/mocks.py:80]: OK I'm going to stop writing question marks. Sort the docstrings out!!

File: tests/cloud/pub_sub/mocks.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    @staticmethod
+    def subscription_path(project_name, subscription_name):
+        return f"projects/{project_name}/subscriptions/{subscription_name}"
+
+
+class MockPullResponse:
+    """A mock PullResponse that is returned by `MockSubscriber.pull`.
+
+    :param iter|None received_messages:
+    :return None:
```
Comment (thclark) [inline tests/cloud/pub_sub/mocks.py:108]: Add param description

**Review by thclark — APPROVED** [review]
This looks really good, so I'll approve to enable you to go forth and merge - but sort the dosctrings please!

File: tests/cloud/pub_sub/mocks.py (hunk 1 above)
Comment (cortadocodes) [reply tests/cloud/pub_sub/mocks.py:14]: Ah I forgot to update them after I based them on another class!

**Pull request merged**